[INTERNAL PR] Dotnet Counters Unification

kkeirstead opened the pull request:
Internal PR to solidify things before bringing this code to the diagnostics repo. Note that I expect some things still need to be tweaked, but I wanted to put this up instead of just continuing to polish it. I also plan to do another round of manual testing on this when it's in a "done" state - I did manual testing to find some minor differences prior to the `shared` session logic, so just to be safe I'll likely do another sanity check. There's also an accompanying PR for `dotnet-monitor` with a few small changes [here](https://github.com/kkeirstead/dotnet-monitor/pull/212).

wiktork committed:
Unify counters

wiktork committed:
fixup merge

kkeirstead committed:
Switched to using AggregatePercentilePayload.

kkeirstead committed:
Changes to AggregatePercentilePayload

kkeirstead committed:
Bug fixes to improve consistency with existing dotnet counters tool

kkeirstead committed:
Adjustments to support shared sessions - this has not been tested.

kkeirstead committed:
Bug fix.

kkeirstead committed:
Merged with main.

hoyosjs committed:
Update analyzers (#4188)

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4190)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4153)

This pull request updates the following dependencies

[marker]: <> (Begin:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)
## From https://github.com/dotnet/runtime
- **Subscription**: e4bfb556-e13c-47f6-eb5a-08d8e4d5099b
- **Build**: 20230825.9
- **Date Produced**: August 26, 2023 2:22:36 AM UTC
- **Commit**: d379e3a1fe89d463ae55b86525fbd7713e983d9e
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.NETCore.App.Runtime.win-x64**: [from 8.0.0-rc.1.23410.15
to 8.0.0-rc.2.23425.9][13]
- **VS.Redist.Common.NetCore.SharedFramework.x64.8.0**: [from
8.0.0-rc.1.23410.15 to 8.0.0-rc.2.23425.9][13]

[13]: https://github.com/dotnet/runtime/compare/786b9872ad...d379e3a1fe

[DependencyUpdate]: <> (End)


[marker]: <> (End:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Mike McLaughlin <mikem@microsoft.com>

am11 committed:
Replace alpine with linux-musl (#4189)

Missed that during the eng/native sync, we made these replacements
upstream in
https://github.com/dotnet/runtime/commit/9fc5fcb0af7d068298bb1950e8d00ce26a70456c
to keep things distro-neutral.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4192)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/arcade (#4193)

[main] Update dependencies from dotnet/arcade

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/installer (#4194)

[main] Update dependencies from dotnet/installer


 - Update Versions.props

lateralusX committed:
Extend dotnet-gcdump support for mobile and dotnet-dsrouter scenarios. (#4081)

.net8 adds support for dotnet-gcdump on Mono, meaning dotnet-gcdump will
be used targeting mobile platforms. Currently dotnet-gcdump doesn't have
support needed since it can only connect using pid or process name and
has logic to make sure the process connected to has the same pid as the
-p argument passed to dotnet-gcdump.

On mobile platforms, runtime is running on device and communicates using
TCP/IP (over loopback interface, using Android adb port forwarding or
usbmux on iOS). All logic related to communicate with devices on
different platforms is handled by dotnet-dsrouter, that expose the same
IPC channels as a normal runtime would do, meaning most diagnostic tools
can connect to dotnet-dsrouter that will route all communication to/from
device. Tools like dotnet-trace can leverage
--diagnostic-port=<ipc>,connect instead of pid to connect to a named IPC
channel on dotnet-dsrouter (routed to a TCP/IP port on device). It is
also possible to connect directly towards dotnet-dsrouters pid since it
will masquerade like a regular runtime, but it will not alter the pid
passed in EventPipe messages, meaning that dotnet.gcdump's pid checks
currently breaks that scenario.

This PR extends diagnostic tooling support for mobile and
dotnet-dsrouter scenarios.

* Add support for --diagnostic-port argument in dotnet-gcdump, but only
support connect mode, since listen mode (reverse diagnostic server) is
mainly for startup tracing where GC dump is not full supported.
* Add support for new command, convert, in dotnet-gcdump that can take a
nettrace file as input and convert it into a gcdump file. Can be useful
if GC dumps have been captured by tools like dotnet-trace, meaning that
dotnet-gcdump will be able to convert it into a gcdump.
* Add ability to tell diagnostic tools currently supporting
--diagnostic-port command that process in -p is a dsrouter process. This
will make the tools construct a default IPC channel used by dsrouter
based on the -p parameter, simplify the connect scenarios against
dsrouter from tools a lot, since it can use the default IPC channel
setup by dsrouter.
* Always setup default IPC server channel in dsrouter if no specific
ipcs argument has been set.
* Fix dsrouter ctrl-c shutdown issue + additional error logging for
diagnostic.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4195)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/symstore (#4196)

[main] Update dependencies from dotnet/symstore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4197)

This pull request updates the following dependencies

[marker]: <> (Begin:8fefa124-13dd-4c66-7dae-08d9c02d7834)
## From https://github.com/dotnet/source-build-reference-packages
- **Subscription**: 8fefa124-13dd-4c66-7dae-08d9c02d7834
- **Build**: 20230828.2
- **Date Produced**: August 28, 2023 1:15:33 PM UTC
- **Commit**: 26ce96327dd346534926c4551f8b8d62a6fc724f
- **Branch**: refs/heads/main

[DependencyUpdate]: <> (Begin)

- **Updates**:
-
**Microsoft.SourceBuild.Intermediate.source-build-reference-packages**:
[from 8.0.0-alpha.1.23424.1 to 8.0.0-alpha.1.23428.2][1]

[1]:
https://github.com/dotnet/source-build-reference-packages/compare/93c23409e6...26ce96327d

[DependencyUpdate]: <> (End)


[marker]: <> (End:8fefa124-13dd-4c66-7dae-08d9c02d7834)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

lateralusX committed:
Add default iOS/Android commands to dotnet-dsrouter. (#4090)

dotnet-dsrouter can be a little hard to configure for mobile use cases
since it needs a number of arguments, both to setup its local IPC
client|server and the corresponding TCP client|server and what arguments
to use depends on what mobile platform and scenario the user runs.

There are currently a number of different scenarios described in
different sources of documentation:

Runtime docs:


https://github.com/dotnet/runtime/blob/main/docs/design/mono/diagnostics-tracing.md

iOS SDK docs:

https://github.com/xamarin/xamarin-macios/wiki/Profiling

Android SDK docs:


https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/tracing.md

They all fall into a number of predefined scenarios and a number of
"default" parameters that should be used.

This PR creates 4 new commands in dotnet-dsrouter to explicitly use the
defaults described in the documentation, ios-sim, ios, android-emu,
android. They all fallback to default IPC server listener for dsrouter
and in order to make that simpler in use with diagnostic tooling,
changes done in https://github.com/dotnet/diagnostics/pull/4081 is also
needed to simplify the process.

So lets take an example form the docs, running an app on iOS simulator.
Before this PR the following dsrouter command needs to be run:

```
dotnet-dsrouter client-server -ipcc ~/my-sim-port -tcps 127.0.0.1:9000

launch app with DOTNET_DiagnosticPorts=127.0.0.1:9000

dotnet-trace collect --diagnostic-port ~/my-sim-port --format speedscope
```

With this PR (and https://github.com/dotnet/diagnostics/pull/4081) the
above will look like:

```
dotnet-dsrouter ios-sim

launch app with DOTNET_DiagnosticPorts=127.0.0.1:9000

dotnet-trace collect -p:<dsrouter pid> --format speedscope
```

dontet-dsrouter will output both its pid as well as what
DOTNET_DiagnosticPorts that can be used to connect to it on startup.

Using a different mobile platform/scenario, like android emulator is
pretty much identical, just switch ios-sim to android-emu.
dotnet-dsrouter will output the exact DOTNET_DiagnosticPorts that should
be used with any of the configured scenarios.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4200)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4202)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from microsoft/clrmd (#4201)

This pull request updates the following dependencies

[marker]: <> (Begin:cb58fe07-ae24-4e73-0e84-08d8e40a189f)
## From https://github.com/microsoft/clrmd
- **Subscription**: cb58fe07-ae24-4e73-0e84-08d8e40a189f
- **Build**: 20230829.1
- **Date Produced**: August 29, 2023 8:58:43 PM UTC
- **Commit**: 0c46ba9ce162de1cfc124379c0d9cde99ed4206f
- **Branch**: refs/heads/main

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.Diagnostics.Runtime**: [from 3.0.442202 to 3.0.442901][1]
- **Microsoft.Diagnostics.Runtime.Utilities**: [from 3.0.442202 to
3.0.442901][1]

[1]: https://github.com/microsoft/clrmd/compare/c7ec730380...0c46ba9ce1

[DependencyUpdate]: <> (End)


[marker]: <> (End:cb58fe07-ae24-4e73-0e84-08d8e40a189f)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4203)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4204)

[main] Update dependencies from dotnet/runtime

davmason committed:
Spec for GetProcessInfo3 IPC command (#3476)

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4206)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4205)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4207)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4208)

[main] Update dependencies from dotnet/source-build-reference-packages

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4210)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/arcade (#4211)

[main] Update dependencies from dotnet/arcade

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/installer (#4212)

[main] Update dependencies from dotnet/installer


 - Update Versions.props

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/symstore (#4214)

[main] Update dependencies from dotnet/symstore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4213)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4216)

[main] Update dependencies from dotnet/source-build-reference-packages

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4215)

[main] Update dependencies from dotnet/aspnetcore

kkeirstead committed:
Clean up before PR

kkeirstead committed:
Small adjustments.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4219)

This pull request updates the following dependencies

[marker]: <> (Begin:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)
## From https://github.com/dotnet/runtime
- **Subscription**: e4bfb556-e13c-47f6-eb5a-08d8e4d5099b
- **Build**: 20230907.7
- **Date Produced**: September 7, 2023 7:21:12 PM UTC
- **Commit**: da3500bb02343b1d0424c74ccdddbc592b5b3f4f
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.NETCore.App.Runtime.win-x64**: [from 8.0.0-rc.2.23431.9 to
8.0.0-rc.2.23457.7][5]
- **VS.Redist.Common.NetCore.SharedFramework.x64.8.0**: [from
8.0.0-rc.2.23431.9 to 8.0.0-rc.2.23457.7][5]

[5]: https://github.com/dotnet/runtime/compare/3c48925a6c...da3500bb02

[DependencyUpdate]: <> (End)


[marker]: <> (End:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Mike McLaughlin <mikem@microsoft.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4218)

This pull request updates the following dependencies

[marker]: <> (Begin:319094f3-ed78-47c4-53e7-08d8e409d87d)
## From https://github.com/dotnet/aspnetcore
- **Subscription**: 319094f3-ed78-47c4-53e7-08d8e409d87d
- **Build**: 20230910.4
- **Date Produced**: September 10, 2023 6:20:35 PM UTC
- **Commit**: a413f2e0f8ec25fc03fc43955b3a70dc7a65e0b0
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.AspNetCore.App.Ref**: [from 8.0.0-rc.2.23455.8 to
8.0.0-rc.2.23460.4][5]
- **Microsoft.AspNetCore.App.Ref.Internal**: [from 8.0.0-rc.2.23455.8 to
8.0.0-rc.2.23460.4][5]

[5]:
https://github.com/dotnet/aspnetcore/compare/2772a78349...a413f2e0f8

[DependencyUpdate]: <> (End)


[marker]: <> (End:319094f3-ed78-47c4-53e7-08d8e409d87d)

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Mike McLaughlin <mikem@microsoft.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4221)

This pull request updates the following dependencies

[marker]: <> (Begin:8fefa124-13dd-4c66-7dae-08d9c02d7834)
## From https://github.com/dotnet/source-build-reference-packages
- **Subscription**: 8fefa124-13dd-4c66-7dae-08d9c02d7834
- **Build**: 20230907.1
- **Date Produced**: September 7, 2023 9:11:09 PM UTC
- **Commit**: 18302345498a62222dd3c236b3d1d140dbaed43d
- **Branch**: refs/heads/main

[DependencyUpdate]: <> (Begin)

- **Updates**:
-
**Microsoft.SourceBuild.Intermediate.source-build-reference-packages**:
[from 8.0.0-alpha.1.23455.3 to 8.0.0-alpha.1.23457.1][4]

[4]:
https://github.com/dotnet/source-build-reference-packages/compare/75ec14a961...1830234549

[DependencyUpdate]: <> (End)


[marker]: <> (End:8fefa124-13dd-4c66-7dae-08d9c02d7834)

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Mike McLaughlin <mikem@microsoft.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/installer (#4222)

This pull request updates the following dependencies

[marker]: <> (Begin:638f1194-0c1a-4d47-eb59-08d8e4d5099b)
## From https://github.com/dotnet/installer
- **Subscription**: 638f1194-0c1a-4d47-eb59-08d8e4d5099b
- **Build**: 20230910.1
- **Date Produced**: September 11, 2023 3:47:52 AM UTC
- **Commit**: 7022131097dd0dc84a45614b7afb0082cf121925
- **Branch**: refs/heads/release/8.0.1xx

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.Dotnet.Sdk.Internal**: [from 8.0.100-rc.2.23454.1 to
8.0.100-rc.2.23460.1][1]

[1]: https://github.com/dotnet/installer/compare/d511823dcc...7022131097

[DependencyUpdate]: <> (End)


[marker]: <> (End:638f1194-0c1a-4d47-eb59-08d8e4d5099b)

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Mike McLaughlin <mikem@microsoft.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4225)

[main] Update dependencies from dotnet/source-build-reference-packages

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4223)

This pull request updates the following dependencies

[marker]: <> (Begin:319094f3-ed78-47c4-53e7-08d8e409d87d)
## From https://github.com/dotnet/aspnetcore
- **Subscription**: 319094f3-ed78-47c4-53e7-08d8e409d87d
- **Build**: 20230911.20
- **Date Produced**: September 12, 2023 1:52:29 AM UTC
- **Commit**: 76ec8ce3302c12c49102acbd34dd9c393d31a3db
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.AspNetCore.App.Ref**: [from 8.0.0-rc.2.23460.4 to
8.0.0-rc.2.23461.20][1]
- **Microsoft.AspNetCore.App.Ref.Internal**: [from 8.0.0-rc.2.23460.4 to
8.0.0-rc.2.23461.20][1]

[1]:
https://github.com/dotnet/aspnetcore/compare/a413f2e0f8...76ec8ce330

[DependencyUpdate]: <> (End)


[marker]: <> (End:319094f3-ed78-47c4-53e7-08d8e409d87d)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4224)

This pull request updates the following dependencies

[marker]: <> (Begin:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)
## From https://github.com/dotnet/runtime
- **Subscription**: e4bfb556-e13c-47f6-eb5a-08d8e4d5099b
- **Build**: 20230911.29
- **Date Produced**: September 12, 2023 8:05:35 AM UTC
- **Commit**: 267b39201599f6493b1d8f23874fbcca9e8f6c96
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.NETCore.App.Runtime.win-x64**: [from 8.0.0-rc.2.23457.7 to
8.0.0-rc.2.23461.29][1]
- **VS.Redist.Common.NetCore.SharedFramework.x64.8.0**: [from
8.0.0-rc.2.23457.7 to 8.0.0-rc.2.23461.29][1]

[1]: https://github.com/dotnet/runtime/compare/da3500bb02...267b392015

[DependencyUpdate]: <> (End)


[marker]: <> (End:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4227)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4226)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4228)

[main] Update dependencies from dotnet/source-build-reference-packages

wiktork commented on src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CounterPayload.cs:
```diff
@@ -73,43 +53,88 @@ internal class CounterPayload : ICounterPayload
+        public int Series { get; }
+    }
+
+    internal sealed class StandardCounterPayload : CounterPayload
```
I'm wondering if StandardCounter is not the right name here.

kkeirstead replied:
Switched this to EventCounterPayload - this seemed more descriptive, but this can be changed to something else.

wiktork commented on src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CounterPayloadExtensions.cs:
```diff
@@ -1,21 +1,46 @@
+
+        private static string GetUnit(string unit, DisplayRenderingMode displayRenderingMode)
+        {
+            if (displayRenderingMode == DisplayRenderingMode.DotnetCounters && string.Equals(unit, "count", StringComparison.OrdinalIgnoreCase))
```
nit: Make some consts for the literals here.

kkeirstead replied:
This is a rare case where I think that might slightly obscure readability since the strings explain themselves better than named constants and they're only used once - currently planning to leave as-is.

wiktork commented on src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/ICounterPayload.cs:
```diff
@@ -41,6 +41,10 @@ internal interface ICounterPayload
+
+        bool IsMeter { get; }
+
+        int Series { get; }
```
I was a bit torn on adding Series into the counter payloads. I found it necessary for the unification, but perhaps we can plumb this value somewhere else into CounterMonitor.

wiktork commented on src/Tools/dotnet-counters/CounterMonitor.cs:
```diff
@@ -6,45 +6,32 @@
-        private int _interval;
         private CounterSet _counterList;
-        private CancellationToken _ct;
+        private CancellationToken _ct; // We aren't using this, but we do pass "ct" to Start -> should we scrap this, or just reference _ct in Start?
```
Let's remove the field if we are not using it anywhere else. Parameters are a lot easier to track than fields

wiktork commented on src/Tools/dotnet-counters/Exporters/ICounterRenderer.cs:
```diff
@@ -1,16 +1,18 @@
-        void Initialize();
-        void EventPipeSourceConnected();
-        void ToggleStatus(bool paused);
+        void Initialize(); //Maps to started?
```
Maybe let's clean up these comments: a clear mapping between ICounterLogger methods and ICounterRenderer methods

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4229)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4230)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4231)

[main] Update dependencies from dotnet/source-build-reference-packages

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4232)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4233)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4234)

This pull request updates the following dependencies

[marker]: <> (Begin:8fefa124-13dd-4c66-7dae-08d9c02d7834)
## From https://github.com/dotnet/source-build-reference-packages
- **Subscription**: 8fefa124-13dd-4c66-7dae-08d9c02d7834
- **Build**: 20230914.1
- **Date Produced**: September 14, 2023 4:36:51 PM UTC
- **Commit**: 41fe34a0b95c36e7683504852b708e538a13e36f
- **Branch**: refs/heads/main

[DependencyUpdate]: <> (Begin)

- **Updates**:
-
**Microsoft.SourceBuild.Intermediate.source-build-reference-packages**:
[from 9.0.0-alpha.1.23463.2 to 9.0.0-alpha.1.23464.1][1]

[1]:
https://github.com/dotnet/source-build-reference-packages/compare/00e23a41d7...41fe34a0b9

[DependencyUpdate]: <> (End)


[marker]: <> (End:8fefa124-13dd-4c66-7dae-08d9c02d7834)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4235)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4236)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4237)

[main] Update dependencies from dotnet/source-build-reference-packages

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4238)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/arcade (#4241)

[main] Update dependencies from dotnet/arcade

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/installer (#4242)

[main] Update dependencies from dotnet/installer


 - Update Versions.props

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4243)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from microsoft/clrmd (#4244)

[main] Update dependencies from microsoft/clrmd

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4246)

[main] Update dependencies from dotnet/source-build-reference-packages

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/symstore (#4245)

This pull request updates the following dependencies

[marker]: <> (Begin:678f7c5b-6647-4e77-0d75-08d8e40a4c7c)
## From https://github.com/dotnet/symstore
- **Subscription**: 678f7c5b-6647-4e77-0d75-08d8e40a4c7c
- **Build**: 20230918.1
- **Date Produced**: September 18, 2023 5:42:54 PM UTC
- **Commit**: 8cc6f03fdbd9c79f0bf9ffbe0a788dca1a81348a
- **Branch**: refs/heads/main

[DependencyUpdate]: <> (Begin)

- **Updates**:
  - **Microsoft.SymbolStore**: [from 1.0.445401 to 1.0.446801][1]

[1]: https://github.com/dotnet/symstore/compare/4827cdd43b...8cc6f03fdb

[DependencyUpdate]: <> (End)


[marker]: <> (End:678f7c5b-6647-4e77-0d75-08d8e40a4c7c)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4248)

[main] Update dependencies from dotnet/runtime

JongHeonChoi committed:
[RISC-V] Set the HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES_EXITCOD value to 0 in RISCV64 (#4249)

https://github.com/dotnet/runtime/pull/92324

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4251)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4250)

[main] Update dependencies from dotnet/aspnetcore

kkeirstead committed:
Small changes.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4252)

This pull request updates the following dependencies

[marker]: <> (Begin:319094f3-ed78-47c4-53e7-08d8e409d87d)
## From https://github.com/dotnet/aspnetcore
- **Subscription**: 319094f3-ed78-47c4-53e7-08d8e409d87d
- **Build**: 20230921.8
- **Date Produced**: September 21, 2023 10:39:05 PM UTC
- **Commit**: cca9840c7d56a44dac424493a94365c7ac5f1427
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.AspNetCore.App.Ref**: [from 8.0.0-rtm.23470.10 to
8.0.0-rtm.23471.8][1]
- **Microsoft.AspNetCore.App.Ref.Internal**: [from 8.0.0-rtm.23470.10 to
8.0.0-rtm.23471.8][1]

[1]:
https://github.com/dotnet/aspnetcore/compare/470af14260...cca9840c7d

[DependencyUpdate]: <> (End)


[marker]: <> (End:319094f3-ed78-47c4-53e7-08d8e409d87d)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

kkeirstead committed:
Fixes small filtering bug with instrumentation started

kkeirstead committed:
Reintroduced display name for aggregate percentile payload

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4255)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4256)

[main] Update dependencies from dotnet/source-build-reference-packages

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4257)

This pull request updates the following dependencies

[marker]: <> (Begin:319094f3-ed78-47c4-53e7-08d8e409d87d)
## From https://github.com/dotnet/aspnetcore
- **Subscription**: 319094f3-ed78-47c4-53e7-08d8e409d87d
- **Build**: 20230924.1
- **Date Produced**: September 24, 2023 4:06:30 PM UTC
- **Commit**: d8a3aa870862d01a8b3a3235f8dd7fc951dd8ff4
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.AspNetCore.App.Ref**: [from 8.0.0-rtm.23471.8 to
8.0.0-rtm.23474.1][1]
- **Microsoft.AspNetCore.App.Ref.Internal**: [from 8.0.0-rtm.23471.8 to
8.0.0-rtm.23474.1][1]

[1]:
https://github.com/dotnet/aspnetcore/compare/cca9840c7d...d8a3aa8708

[DependencyUpdate]: <> (End)


[marker]: <> (End:319094f3-ed78-47c4-53e7-08d8e409d87d)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4262)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from microsoft/clrmd (#4247)

[main] Update dependencies from microsoft/clrmd

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4264)

[main] Update dependencies from dotnet/source-build-reference-packages

thaystg committed:
Adding more parameters to connect to mono runtime (#4261)

Now it's tested on all platforms: Windows, Linux and Mac, connecting to
iOS and Android, devices and emulators.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4263)

This pull request updates the following dependencies

[marker]: <> (Begin:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)
## From https://github.com/dotnet/runtime
- **Subscription**: e4bfb556-e13c-47f6-eb5a-08d8e4d5099b
- **Build**: 20230925.7
- **Date Produced**: September 25, 2023 8:30:46 PM UTC
- **Commit**: a3d5300baee03169bf737cfc4e9b1f7dbece4d05
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.NETCore.App.Runtime.win-x64**: [from 8.0.0-rtm.23472.12 to
8.0.0-rtm.23475.7][1]
- **VS.Redist.Common.NetCore.SharedFramework.x64.8.0**: [from
8.0.0-rtm.23472.12 to 8.0.0-rtm.23475.7][1]

[1]: https://github.com/dotnet/runtime/compare/e95228dbcf...a3d5300bae

[DependencyUpdate]: <> (End)


[marker]: <> (End:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/installer (#4258)

[main] Update dependencies from dotnet/installer


 - Update Versions.props

 - Update Versions.props

Commit:
Update Microsoft.DiaSymReader.Native version (#4265)

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4266)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4267)

This pull request updates the following dependencies

[marker]: <> (Begin:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)
## From https://github.com/dotnet/runtime
- **Subscription**: e4bfb556-e13c-47f6-eb5a-08d8e4d5099b
- **Build**: 20230926.15
- **Date Produced**: September 27, 2023 7:00:19 AM UTC
- **Commit**: 0933e300f0c0647a15a0433f1a3b07bcab9882f4
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.NETCore.App.Runtime.win-x64**: [from 8.0.0-rtm.23475.7 to
8.0.0-rtm.23476.15][1]
- **VS.Redist.Common.NetCore.SharedFramework.x64.8.0**: [from
8.0.0-rtm.23475.7 to 8.0.0-rtm.23476.15][1]

[1]: https://github.com/dotnet/runtime/compare/a3d5300bae...0933e300f0

[DependencyUpdate]: <> (End)


[marker]: <> (End:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4271)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4270)

[main] Update dependencies from dotnet/aspnetcore

kkeirstead committed:
Partial PR Feedback - has not been tested, and needs more changes.

cshung committed:
Change Frozen object heap to NonGC heap in !EEHeap output (#4269)

kkeirstead committed:
More PR Feedback.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4273)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4275)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from microsoft/clrmd (#4274)

This pull request updates the following dependencies

[marker]: <> (Begin:cb58fe07-ae24-4e73-0e84-08d8e40a189f)
## From https://github.com/microsoft/clrmd
- **Subscription**: cb58fe07-ae24-4e73-0e84-08d8e40a189f
- **Build**: 20230928.1
- **Date Produced**: September 28, 2023 10:49:03 PM UTC
- **Commit**: 10e87fb192d41e22d5fd0fd54f02ed011726e991
- **Branch**: refs/heads/main

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.Diagnostics.Runtime**: [from 3.0.447501 to 3.0.447801][1]
- **Microsoft.Diagnostics.Runtime.Utilities**: [from 3.0.447501 to
3.0.447801][1]

[1]: https://github.com/microsoft/clrmd/compare/6d7c5a7288...10e87fb192

[DependencyUpdate]: <> (End)


[marker]: <> (End:cb58fe07-ae24-4e73-0e84-08d8e40a189f)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

Commit:
Fix build breaks (#4276)

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4277)

[main] Update dependencies from dotnet/aspnetcore

eduardo-vp committed:
Update SOS to show the relevant information for the !ThreadPool command when using the Windows thread pool (#4160)

Uses the properties exposed in
https://github.com/microsoft/clrmd/pull/1175 to show relevant Windows
thread pool information if it's enabled

---------

Co-authored-by: Mike McLaughlin <mikem@microsoft.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from microsoft/clrmd (#4278)

This pull request updates the following dependencies

[marker]: <> (Begin:cb58fe07-ae24-4e73-0e84-08d8e40a189f)
## From https://github.com/microsoft/clrmd
- **Subscription**: cb58fe07-ae24-4e73-0e84-08d8e40a189f
- **Build**: 20230929.1
- **Date Produced**: September 29, 2023 3:41:14 PM UTC
- **Commit**: 547545a301475a7cfd23ce8c2f6e24eddf55d83e
- **Branch**: refs/heads/main

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.Diagnostics.Runtime**: [from 3.0.447801 to 3.0.447901][1]
- **Microsoft.Diagnostics.Runtime.Utilities**: [from 3.0.447801 to
3.0.447901][1]

[1]: https://github.com/microsoft/clrmd/compare/10e87fb192...547545a301

[DependencyUpdate]: <> (End)


[marker]: <> (End:cb58fe07-ae24-4e73-0e84-08d8e40a189f)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4280)

[main] Update dependencies from dotnet/aspnetcore

lateralusX committed:
Improve dotnet-dsrouter log levels and info. (#4199)

* Add support for -v none, suppressing all logging making
dotnet-dsrouter quiet.
* Add support for --info option for new iOS/Android preconfig commands,
outputs details on how to configure app and diagnostic tools to connect
to current running dsrouter instance.
*  Improve error handling when failing to find or execute adb binary.
*  Specify and validate supported verbose logging levels.

dotnet-maestro[bot] committed:
[main] Update dependencies from microsoft/clrmd (#4281)

[main] Update dependencies from microsoft/clrmd

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4282)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/installer (#4283)

[main] Update dependencies from dotnet/installer


 - Update Versions.props

Commit:
Remove the symstore dependency from Microsoft.Diagnostics.DebugServices (#4279)

This is needed to reduce the conflicts/dependencies on loadable
extensions.

Commit:
Add the "crashinfo" command for NativeAOT. (#4125)

Reads the JSON crash info from the triage buffer in the NativeAOT
runtime.

Adds a ICrashInfoService and SpecialDiagInfo helper class.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4289)

[main] Update dependencies from dotnet/source-build-reference-packages

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4286)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4288)

This pull request updates the following dependencies

[marker]: <> (Begin:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)
## From https://github.com/dotnet/runtime
- **Subscription**: e4bfb556-e13c-47f6-eb5a-08d8e4d5099b
- **Build**: 20231002.11
- **Date Produced**: October 3, 2023 11:06:54 AM UTC
- **Commit**: ef6283ac0a14c78d9e9fef4841545099bd7ad12b
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.NETCore.App.Runtime.win-x64**: [from 8.0.0-rtm.23478.17 to
8.0.0-rtm.23502.11][1]
- **VS.Redist.Common.NetCore.SharedFramework.x64.8.0**: [from
8.0.0-rtm.23478.17 to 8.0.0-rtm.23502.11][1]

[1]: https://github.com/dotnet/runtime/compare/b20f704cc0...ef6283ac0a

[DependencyUpdate]: <> (End)


[marker]: <> (End:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

kkeirstead committed:
PR Feedback.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4291)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4292)

This pull request updates the following dependencies

[marker]: <> (Begin:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)
## From https://github.com/dotnet/runtime
- **Subscription**: e4bfb556-e13c-47f6-eb5a-08d8e4d5099b
- **Build**: 20231003.15
- **Date Produced**: October 4, 2023 5:51:18 AM UTC
- **Commit**: a84f8ffbf5d597b8a91e893a1f413466de017a68
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.NETCore.App.Runtime.win-x64**: [from 8.0.0-rtm.23502.11 to
8.0.0-rtm.23503.15][1]
- **VS.Redist.Common.NetCore.SharedFramework.x64.8.0**: [from
8.0.0-rtm.23502.11 to 8.0.0-rtm.23503.15][1]

[1]: https://github.com/dotnet/runtime/compare/ef6283ac0a...a84f8ffbf5

[DependencyUpdate]: <> (End)


[marker]: <> (End:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

Commit:
Command groups to support duplicate command names and better help support (#4285)

Add extension load testing that includes duplicate command names.

Add internal CommandGroup class to CommandService

dotnet-dump analyze -c/--command will exit on any parsing errors or
exceptions in the command

command help not displayed on parsing errors/invalid options

Add INIT_API_CHECK_MANAGED native SOS command macro

Rename clrmodules command to assemblies and keep clrmodules as an alias

Support static HelpInvoke/FilterInvoke methods

Add command service testing

Better command service help interface. Returns help text instead of
printing it on the console directly. ICommandService.DisplayHelp() =>
GetHelpText(). Better help sorting.

Fix some SOS scripts

Return error code for command line option errors in native SOS

Load next to executing assembly first when hosted under desktop
Framework

Test using "clrthreads" instead of "Threads"

Replace testing "u" with "clru"

Add more general command filter mechanism. Add FilterType property to
Command attribute. Remove OS filter command flags.

NativeAddressHelper service work

Move Windows managed command stubs to separate file

Add SOS.Hosting services to SOS.Extensions for better native command
help. This requires the special ManagedOnlyCommandFilter service to
prevent recursion of the C++ commands. On Windows this recursion is from
"C++ command" -> "checking if managed version" -> "executing the command
in SOS.Hosting" -> "C++ command".

Help is now uniform across managed/native, alphabetized and filtered by
the current runtime.

dotnet-maestro[bot] committed:
[main] Update dependencies from microsoft/clrmd (#4287)

This pull request updates the following dependencies

[marker]: <> (Begin:cb58fe07-ae24-4e73-0e84-08d8e40a189f)
## From https://github.com/microsoft/clrmd
- **Subscription**: cb58fe07-ae24-4e73-0e84-08d8e40a189f
- **Build**: 20231004.2
- **Date Produced**: October 4, 2023 10:57:18 PM UTC
- **Commit**: 10828baeeb7a144414259d9c00a9923c2fcb61e7
- **Branch**: refs/heads/main

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.Diagnostics.Runtime**: [from 3.0.450101 to 3.1.450402][3]
- **Microsoft.Diagnostics.Runtime.Utilities**: [from 3.0.450101 to
3.1.450402][3]

[3]: https://github.com/microsoft/clrmd/compare/1c8f0f18ea...10828baeeb

[DependencyUpdate]: <> (End)


[marker]: <> (End:cb58fe07-ae24-4e73-0e84-08d8e40a189f)

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4294)

This pull request updates the following dependencies

[marker]: <> (Begin:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)
## From https://github.com/dotnet/runtime
- **Subscription**: e4bfb556-e13c-47f6-eb5a-08d8e4d5099b
- **Build**: 20231004.8
- **Date Produced**: October 4, 2023 10:14:40 PM UTC
- **Commit**: 22b8a5665f5725a2c7bb09cfbe88f2cdc9847c1a
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.NETCore.App.Runtime.win-x64**: [from 8.0.0-rtm.23503.15 to
8.0.0-rtm.23504.8][1]
- **VS.Redist.Common.NetCore.SharedFramework.x64.8.0**: [from
8.0.0-rtm.23503.15 to 8.0.0-rtm.23504.8][1]

[1]: https://github.com/dotnet/runtime/compare/a84f8ffbf5...22b8a5665f

[DependencyUpdate]: <> (End)


[marker]: <> (End:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4293)

This pull request updates the following dependencies

[marker]: <> (Begin:319094f3-ed78-47c4-53e7-08d8e409d87d)
## From https://github.com/dotnet/aspnetcore
- **Subscription**: 319094f3-ed78-47c4-53e7-08d8e409d87d
- **Build**: 20231004.12
- **Date Produced**: October 4, 2023 8:08:32 PM UTC
- **Commit**: 91314b6bde6e9823f848fb37100aea5df4e44c99
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.AspNetCore.App.Ref**: [from 8.0.0-rtm.23503.8 to
8.0.0-rtm.23504.12][1]
- **Microsoft.AspNetCore.App.Ref.Internal**: [from 8.0.0-rtm.23503.8 to
8.0.0-rtm.23504.12][1]

[1]:
https://github.com/dotnet/aspnetcore/compare/2d2f880867...91314b6bde

[DependencyUpdate]: <> (End)


[marker]: <> (End:319094f3-ed78-47c4-53e7-08d8e409d87d)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

Commit:
Use new ModuleInfo.TryCreateResourceRoot in DataReader impl (#4295)

Some test fixes to make them more reliable.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4297)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4299)

This pull request updates the following dependencies

[marker]: <> (Begin:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)
## From https://github.com/dotnet/runtime
- **Subscription**: e4bfb556-e13c-47f6-eb5a-08d8e4d5099b
- **Build**: 20231006.12
- **Date Produced**: October 7, 2023 3:40:26 AM UTC
- **Commit**: b17a34c818bd5e01fdb9827fea64727a5fc51025
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.NETCore.App.Runtime.win-x64**: [from 8.0.0-rtm.23504.8 to
8.0.0-rtm.23506.12][1]
- **VS.Redist.Common.NetCore.SharedFramework.x64.8.0**: [from
8.0.0-rtm.23504.8 to 8.0.0-rtm.23506.12][1]

[1]: https://github.com/dotnet/runtime/compare/22b8a5665f...b17a34c818

[DependencyUpdate]: <> (End)


[marker]: <> (End:e4bfb556-e13c-47f6-eb5a-08d8e4d5099b)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4298)

This pull request updates the following dependencies

[marker]: <> (Begin:319094f3-ed78-47c4-53e7-08d8e409d87d)
## From https://github.com/dotnet/aspnetcore
- **Subscription**: 319094f3-ed78-47c4-53e7-08d8e409d87d
- **Build**: 20231006.5
- **Date Produced**: October 7, 2023 1:42:18 AM UTC
- **Commit**: 6bd37340c54e3b9690102886afca6198a461cb3e
- **Branch**: refs/heads/release/8.0

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.AspNetCore.App.Ref**: [from 8.0.0-rtm.23505.1 to
8.0.0-rtm.23506.5][1]
- **Microsoft.AspNetCore.App.Ref.Internal**: [from 8.0.0-rtm.23505.1 to
8.0.0-rtm.23506.5][1]

[1]:
https://github.com/dotnet/aspnetcore/compare/51f80c73e5...6bd37340c5

[DependencyUpdate]: <> (End)


[marker]: <> (End:319094f3-ed78-47c4-53e7-08d8e409d87d)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/arcade (#4300)

[main] Update dependencies from dotnet/arcade


 - Fix build from new analysis checks

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/installer (#4301)

This pull request updates the following dependencies

[marker]: <> (Begin:638f1194-0c1a-4d47-eb59-08d8e4d5099b)
## From https://github.com/dotnet/installer
- **Subscription**: 638f1194-0c1a-4d47-eb59-08d8e4d5099b
- **Build**: 20231006.1
- **Date Produced**: October 6, 2023 9:36:48 PM UTC
- **Commit**: 0ffc9fdc93e578268a09b0dccdc4c3527f4697f3
- **Branch**: refs/heads/release/8.0.1xx

[DependencyUpdate]: <> (Begin)

- **Updates**:
- **Microsoft.Dotnet.Sdk.Internal**: [from 8.0.100-rtm.23479.3 to
8.0.100-rtm.23506.1][1]

[1]: https://github.com/dotnet/installer/compare/6dae8493a5...0ffc9fdc93

[DependencyUpdate]: <> (End)


[marker]: <> (End:638f1194-0c1a-4d47-eb59-08d8e4d5099b)

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Mike McLaughlin <mikem@microsoft.com>

Commit:
Remove SOS's CRT invalid parameter handler (#4303)

SOS does get built dynamically linking to the C++ CRT in release so the
invalid parameter handler that gets installed affects all the code in
the process.

Fixes issue: https://github.com/dotnet/diagnostics/issues/4070

Commit:
Re-enable dumpmt tests in OtherCommands.script (#4304)

Addresses issue #https://github.com/dotnet/diagnostics/issues/3516

dotnet-maestro[bot] committed:
[main] Update dependencies from microsoft/clrmd (#4306)

[main] Update dependencies from microsoft/clrmd

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4305)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/runtime (#4308)

[main] Update dependencies from dotnet/runtime

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/symstore (#4307)

This pull request updates the following dependencies

[marker]: <> (Begin:678f7c5b-6647-4e77-0d75-08d8e40a4c7c)
## From https://github.com/dotnet/symstore
- **Subscription**: 678f7c5b-6647-4e77-0d75-08d8e40a4c7c
- **Build**: 20231009.1
- **Date Produced**: October 9, 2023 4:51:26 PM UTC
- **Commit**: a3b341f9e61c8d8e832c4acfeb5b3a2305e51bcc
- **Branch**: refs/heads/main

[DependencyUpdate]: <> (Begin)

- **Updates**:
  - **Microsoft.SymbolStore**: [from 1.0.446801 to 1.0.450901][1]

[1]: https://github.com/dotnet/symstore/compare/8cc6f03fdb...a3b341f9e6

[DependencyUpdate]: <> (End)


[marker]: <> (End:678f7c5b-6647-4e77-0d75-08d8e40a4c7c)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4309)

This pull request updates the following dependencies

[marker]: <> (Begin:8fefa124-13dd-4c66-7dae-08d9c02d7834)
## From https://github.com/dotnet/source-build-reference-packages
- **Subscription**: 8fefa124-13dd-4c66-7dae-08d9c02d7834
- **Build**: 20231009.2
- **Date Produced**: October 9, 2023 1:15:04 PM UTC
- **Commit**: 8a2f652b1f23d493fcce31b73e829de56df38d5f
- **Branch**: refs/heads/main

[DependencyUpdate]: <> (Begin)

- **Updates**:
-
**Microsoft.SourceBuild.Intermediate.source-build-reference-packages**:
[from 9.0.0-alpha.1.23502.1 to 9.0.0-alpha.1.23509.2][1]

[1]:
https://github.com/dotnet/source-build-reference-packages/compare/05ffbf9df6...8a2f652b1f

[DependencyUpdate]: <> (End)


[marker]: <> (End:8fefa124-13dd-4c66-7dae-08d9c02d7834)

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

Maoni0 committed:
fix doc for fq -allReady (#4310)

fixing some incorrect info for the explaination of the -allReady arg for the !fq cmd.

dotnet-maestro[bot] committed:
[main] Update dependencies from microsoft/clrmd (#4312)

[main] Update dependencies from microsoft/clrmd

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/aspnetcore (#4311)

[main] Update dependencies from dotnet/aspnetcore

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/source-build-reference-packages (#4313)

[main] Update dependencies from dotnet/source-build-reference-packages

kkeirstead committed:
Dotnet Counters + Dotnet Monitor Unification - Automated `Collect` Testing (#4253)

kkeirstead committed:
Merge branch 'main' of https://github.com/dotnet/diagnostics into kkeirstead/unifyCounters5

kkeirstead committed:
Small bug fix for duplicated units on histogram
